app/lib/actions/pulp/repository/ensure_sync_notification.rb

```diff
@@ -0,0 +1,31 @@
+module Actions
+  module Pulp
+    module Repository
+      class EnsureSyncNotification < Pulp::Abstract
+        def run
+          resource = pulp_resources.event_notifier
+          url = SETTINGS[:katello][:post_sync_url]
+          type = Runcible::Resources::EventNotifier::EventTypes::REPO_SYNC_COMPLETE
+          notifs = resource.list
+
+          output[:existing] = []
+          output[:deleted] = []
+
+          # delete any similar tasks with the wrong url (in case it changed)
+          notifs.select { |n| n['event_types'] == [type] }.each do |e|
+            if e['notifier_config']['url'] != url
+              resource.delete(e['id'])
+              output[:deleted] << e
+            else
+              output[:existing] << e
+            end
+          end
+
+          # only create a notifier if one doesn't exist with the correct url
+          exists = notifs.select { |n| n['event_types'] == [type] && n['notifier_config']['url'] == url }
+          resource.create(Runcible::Resources::EventNotifier::NotifierTypes::REST_API, {:url => url}, [type]) if exists.empty?
+        end
+      end
+    end
+  end
+end
```

Comment:
I actually like having the logic in repo.rb as its much simpler to test there.  I'm more in favor of moving as much logic as we can out of the tasks (aside from the logic which plans the tasks).

Reply:
Ok, left in the repo.rb.  Can you take a look now?